lib/msal-core/src/Storage.ts

```diff
@@ -90,18 +90,31 @@ export class Storage {// Singleton
         return results;
     }
 
-    removeAcquireTokenEntries(authorityKey: string, acquireTokenAccountKey: string): void {
+    removeAcquireTokenEntries(): void {
         const storage = window[this.cacheLocation];
         if (storage) {
             let key: string;
             for (key in storage) {
                 if (storage.hasOwnProperty(key)) {
-                    if ((authorityKey !== "" && key.indexOf(authorityKey) > -1) || (acquireTokenAccountKey !== "" && key.indexOf(acquireTokenAccountKey) > -1)) {
-                        this.removeItem(key);
+                    if (key.indexOf(CacheKeys.AUTHORITY) !== -1 || key.indexOf(CacheKeys.ACQUIRE_TOKEN_USER) !== 1) {
+                        const state = key.split(Constants.resourceDelimiter)[1];
```

Comment:
any case where the array does not have element [1]

Reply:
We always store CacheKeys.Authority or CacheKeys.AcquireToken with the resource delim and state. I can add a check though.

Reply:
updated